hilti/runtime/src/init.cc

```diff
@@ -75,6 +77,22 @@ void hilti::rt::done() {
 bool hilti::rt::isInitialized() { return __global_state && __global_state->runtime_is_initialized; }
 
 void hilti::rt::detail::registerModule(HiltiModule module) {
+    const auto* id = module.id;
+    const auto* name = module.name;
+
+    assert(name);
+    assert(id);
+
+    const auto& hilti_modules = globalState()->hilti_modules;
+    if ( hilti_modules.end() !=
+         std::find_if(hilti_modules.begin(), hilti_modules.end(), [&](const HiltiModule& module) {
+             return std::strcmp(module.name, name) == 0 && std::strcmp(module.id, id) == 0;
+         }) ) {
+        HILTI_RT_DEBUG("libhilti", fmt("skipping registration of module %s since the module was registered previously",
+                                       module.name));
+        return;
+    }
```

Comment:
Just an optional suggestion: Personally, I think a plain-old for-loop would be quite a bit more readable ...